modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,595 @@
+		// Allow namespaced keys by use of colon in $key, namespaces must be part of the document
+		$namespace = null;
+		if ( false !== strpos( $key, ':' ) ) {
+			list( $namespace_prefix, $key ) = explode( ':', $key );
+			if ( isset( $doc_namespaces[ $namespace_prefix ] ) ) {
+				$namespace = $doc_namespaces[ $namespace_prefix ];
+			}
+		}
+
+		if ( is_array( $value ) ) {
+			$child = $tree->addChild( $key, null, $namespace );
+			jetpack_sitemap_array_to_simplexml( $value, $child );
+		} else {
+			$tree->addChild( $key, esc_html( $value ), $namespace );
+		}
+	}
+
+	return $tree;
+}
+
+/**
+ * Define an array of attribute value pairs for use inside the root element of an XML document.
+ * Intended for mapping namespace and namespace URI values.
+ * Passes array through jetpack_sitemap_ns for other functions to add their own namespaces
+ *
+ * @return array array of attribute value pairs passed through the jetpack_sitemap_ns filter
+ */
+function jetpack_sitemap_namespaces() {
+	/**
+	 * Filter the attribute value pairs used for namespace and namespace URI mappings.
+	 *
+	 * @since 3.9
+	 *
+	 * @param array $namespaces Associative array with namespaces and namespace URIs.
+	 */
+	return apply_filters( 'jetpack_sitemap_ns', array (
+		'xmlns:xsi'          => 'http://www.w3.org/2001/XMLSchema-instance',
+		'xsi:schemaLocation' => 'http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd',
+		'xmlns'              => 'http://www.sitemaps.org/schemas/sitemap/0.9',
+		// Mobile namespace from http://support.google.com/webmasters/bin/answer.py?hl=en&answer=34648
+		'xmlns:mobile'       => 'http://www.google.com/schemas/sitemap-mobile/1.0',
+		'xmlns:image'        => 'http://www.google.com/schemas/sitemap-image/1.1',
+	) );
+}
+
+/**
+ * Start sitemap XML document, writing its heading and <urlset> tag with namespaces.
+ *
+ * @param $charset string Charset for current XML document.
+ *
+ * @return string
+ */
+function jetpack_sitemap_initstr( $charset ) {
+	$initstr = '<?xml version="1.0" encoding="' . $charset . '"?>' . "\n" . '<!-- generator="jetpack" -->' . "\n";
+	$initstr .= '<urlset';
+	foreach ( jetpack_sitemap_namespaces() as $attribute => $value ) {
+		$initstr .= ' ' . esc_html( $attribute ) . '="' . esc_attr( $value ) . '"';
+	}
+	$initstr .= ' />';
+
+	return $initstr;
+}
+
+/**
+ * Print an XML sitemap conforming to the Sitemaps.org protocol
+ * Outputs an XML list of up to the latest 1000 posts.
+ *
+ * @link http://sitemaps.org/protocol.php Sitemaps.org protocol
+ */
+function jetpack_print_sitemap() {
+	if ( defined( 'JETPACK_SKIP_DEFAULT_SITEMAP' ) && JETPACK_SKIP_DEFAULT_SITEMAP ) {
+		return;
+	}
+	global $wpdb;
+
+	$xml = get_transient( 'jetpack_sitemap' );
+
+	if ( $xml ) {
+		header( 'Content-Type: ' . jetpack_sitemap_content_type(), true );
+		echo $xml;
+		die();
+	}
+
+	$post_types = array ( 'post', 'page' );
+
+	$post_types_in = array ();
+	/**
```

Comment:
@jeherve done. What do you think about dismissing that `$post_types` assignation and instead do it directly in the filter passing the array?